### PR TITLE
[Spec] update specs of voting, transaction_fee, code

### PR DESCRIPTION
**Pull request by xudon9**
### Description
This is a part of spec work for Aptos Framework from MoveBit. The changes are:
- Completed all specs left as TODO in `transaction_fee` and `voting`;
- Some functions in `code` gets specified.
### Test Plan
`aptos move prove -f voting.move`
`aptos move prove -f transaction_fee.move`
`aptos move prove -f code.move`

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/voting.move [hunk 1]
```diff
@@ -364,6 +364,11 @@ module aptos_framework::voting {
         proposal_id: u64,
     ) acquires VotingForum {
         let proposal_state = get_proposal_state<ProposalType>(voting_forum_address, proposal_id);
+
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/voting.move:367]: Using assume with ghost state is not an ideal way to specify aborts_if condition. Could you try to write a spec fun version of get_proposal_state instead? If it is difficult to do so, we can visit this aborts_if condition later.

File: aptos-move/framework/aptos-framework/sources/voting.move [hunk 2]
```diff
@@ -372,7 +377,13 @@ module aptos_framework::voting {
 
         // We need to make sure that the resolution is happening in
         // a separate transaction from the last vote to guard against any potential flashloan attacks.
+
         let resolvable_time = to_u64(*simple_map::borrow(&proposal.metadata, &utf8(RESOLVABLE_TIME_METADATA_KEY)));
+
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/voting.move:382]: Same as above

**Review by rahxephon89 — CHANGES_REQUESTED** [review]

**Review by UIZorrot — COMMENTED** [review]
We have fixed it on the newest commit

File: aptos-move/framework/aptos-framework/sources/voting.spec.move [hunk 3]
```diff
@@ -114,14 +114,23 @@ spec aptos_framework::voting {
         voting_forum_address: address,
         proposal_id: u64,
     ) {
+        
         use aptos_framework::chain_status;
-
+
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/voting.spec.move:119]: remove the trailing whitespaces.

File: aptos-move/framework/aptos-framework/sources/voting.spec.move [hunk 4]
```diff
@@ -164,16 +173,25 @@ spec aptos_framework::voting {
...
     ): u64 {
+        
         use aptos_framework::chain_status;
+        pragma opaque;
         requires chain_status::is_operating(); // Ensures existence of Timestamp
         // Addition of yes_votes and no_votes might overflow.
         pragma addition_overflow_unchecked;
+        
         include AbortsIfNotContainProposalID<ProposalType>;
         // Any way to specify the result?
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/voting.spec.move:193]: what does this mean?

**Review by junkil-park — APPROVED** [review]

**Review by rahxephon89 — APPROVED** [review]

**Pull request merged**